devicetypes/smartthings/testing/simulated-window-shade.src/simulated-window-shade.groovy

```diff
@@ -0,0 +1,183 @@
+			}
+		}
+
+		valueTile("blank", "device.blank", width: 2, height: 2, decoration: "flat") {
+			state "default", label: ""	
+		}
+		valueTile("commandsLabel", "device.commands", width: 6, height: 1, decoration: "flat") {
+			state "default", label: "Commands:"	
+		}
+
+		standardTile("windowShadeOpen", "device.windowShade", width: 2, height: 2, decoration: "flat") {
+			state "default", label: "open", action:"open", icon:"st.Home.home2"
+		}
+		standardTile("windowShadeClose", "device.windowShade", width: 2, height: 2, decoration: "flat") {
+			state "default", label: "close", action:"close", icon:"st.Home.home2"
+		}
+		standardTile("windowShadePause", "device.windowShade", width: 2, height: 2, decoration: "flat") {
+			state "default", label: "pause", action:"pause", icon:"st.Home.home2"
+		}
+		standardTile("windowShadePreset", "device.windowShadePreset", width: 2, height: 2, decoration: "flat") {
+			state "default", label: "preset", action:"presetPosition", icon:"st.Home.home2"
+		}
+
+		valueTile("statesLabel", "device.states", width: 6, height: 1, decoration: "flat") {
+			state "default", label: "State Events:"	
+		}
+
+		standardTile("windowShadePartiallyOpen", "device.windowShade", width: 2, height: 2, decoration: "flat") {
+			state "default", label: "partially open", action:"partiallyOpen", icon:"st.Home.home2"
+		}
+		standardTile("windowShadeOpening", "device.windowShade", width: 2, height: 2, decoration: "flat") {
+			state "default", label: "opening", action:"opening", icon:"st.Home.home2"
+		}
+		standardTile("windowShadeClosing", "device.windowShade", width: 2, height: 2, decoration: "flat") {
+			state "default", label: "closing", action:"closing", icon:"st.Home.home2"
+		}
+		standardTile("windowShadeOpened", "device.windowShade", width: 2, height: 2, decoration: "flat") {
+			state "default", label: "opened", action:"opened", icon:"st.Home.home2"
+		}
+		standardTile("windowShadeClosed", "device.windowShade", width: 2, height: 2, decoration: "flat") {
+			state "default", label: "closed", action:"closed", icon:"st.Home.home2"
+		}
+		standardTile("windowShadeUnknown", "device.windowShade", width: 2, height: 2, decoration: "flat") {
+			state "default", label: "unknown", action:"unknown", icon:"st.Home.home2"
+		}
+
+		main(["windowShade"])
+		details(["windowShade",
+				 "commandsLabel",
+				 "windowShadeOpen", "windowShadeClose", "windowShadePause", "windowShadePreset", "blank", "blank",
+				 "statesLabel",
+				 "windowShadePartiallyOpen", "windowShadeOpening", "windowShadeClosing", "windowShadeOpened", "windowShadeClosed", "windowShadeUnknown"])
+
+	}
+}
+
+def parse(String description) {
+}
+
+def open() {
+	log.debug "open()"
+	opening()
+	runIn(5, "opened")
+}
+
+def close() {
+	log.debug "close()"
+	closing()
+	runIn(5, "closed")
+}
+
+def pause() {
+	log.debug "pause()"
+	partiallyOpen()
+}
+
+def openPartially() {
+	log.debug "openPartially()"
+	opening()
+	runIn(5, "partiallyOpen")
```

Comment:
Could use a preference to control the delay rather than hardcoding it. Might be useful to change it at runtime

Reply:
Added! Eventually I'll update this to use a state machine (or, rather, have a state machine mode).